### PR TITLE
Fixes a semi-rare bee hard delete

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

While unit-testing the nuke with #67144, I discovered that maps with roundstart bees will occasionally hard delete with my setup. So I investigated... 

Queen bees are held within in an item.
The nuke, when detonating, gibs and deletes all mobs on the z-level. 
Bees are on the z-level.
The bee within the bee item is deleted and hard deletes because it lacks reference handling if it were to be deleted. 

So, by adding some extra criteria to ensure "if the bee is deleted, get rid of the item", the hard deletes should stop too.

## Why It's Good For The Game

Stops hard deletes. 

## Changelog

:cl: Melbert
fix: Fixes a rare hard-delete with queen bees. 
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/beekeeping/beebox.dm [hunk 1]
```diff
@@ -190,7 +190,6 @@
 		else
 			to_chat(user, span_warning("The queen bee disappeared! Disappearing bees have been in the news lately..."))
 
-		qdel(qb)
```
Comment (LemonInTheDark) [inline code/modules/hydroponics/beekeeping/beebox.dm:193]: where'd this go?

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 2]
```diff
@@ -290,14 +290,37 @@
...
 	return FALSE
 
+/mob/living/simple_animal/hostile/bee/consider_wakeup()
+	if (beehome && loc == beehome) // If bees are chilling in their nest, they're not actively looking for targets
+		idle = min(100, ++idle)
+		if(idle >= BEE_IDLE_ROAMING && prob(BEE_PROB_GOROAM))
+			toggle_ai(AI_ON)
+			forceMove(beehome.drop_location())
+	else
+		return ..()
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:300]:
```suggestion
/mob/living/simple_animal/hostile/bee/consider_wakeup()
	if (!beehome || loc != beehome)
		return ..()
    // If bees are chilling in their nest, they're not actively looking for targets
    idle = min(100, ++idle)
    if(idle >= BEE_IDLE_ROAMING && prob(BEE_PROB_GOROAM))
        toggle_ai(AI_ON)
        forceMove(beehome.drop_location())
```
early returns

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/beekeeping/beebox.dm (hunk 1 above)
Comment (MrMelbert) [reply code/modules/hydroponics/beekeeping/beebox.dm:193]: the `forceMove()` above calls `Exited()` and deletes it, calling `qdel()` again qdels a qdeleted object

File: code/modules/mob/living/simple_animal/hostile/bees.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/mob/living/simple_animal/hostile/bees.dm:300]: grumble grumble didn't even touch the code just moved it so it made more sense (ok)

File: code/modules/hydroponics/beekeeping/beebox.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/hydroponics/beekeeping/beebox.dm:193]: aight

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 3]
```diff
@@ -290,14 +290,38 @@
 		return TRUE
 	return FALSE
 
+/mob/living/simple_animal/hostile/bee/consider_wakeup()
+	// If bees are chilling in their nest, they're not actively looking for targets.
+	if (!beehome || loc == beehome)
+		return ..()
+
+	idle = min(100, ++idle)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:298]: now that I look closer
```suggestion
	idle = min(100, idle + 1)
```

**Review by LemonInTheDark — APPROVED** [review]

**Comment by Mothblocks** [comment]
I would like to see a new unit test be added that cleanly replicates this hard delete. Ideally you would make this before you write the fix, because step 1 to properly solving an inconsistent test failure is to write a test that reproduces it 100% of the time.

**Comment by MrMelbert** [comment]
it can be replicated fairly easily - if the bee within in qdel'd, it'll hard delete, because it's not reference handled. a test for it should be pretty easy

**Comment by LemonInTheDark** [comment]
There's no point to testing this unless we want to start testing all non init reference pairings

**Comment by Mothblocks** [comment]
Of course we do! We fixed a bug, now we should make sure it doesn't come back again. Having a lot of coverage is a goal worth striving for.

**Comment by Mothblocks** [comment]
I'm going to merge this because CI is failing but definitely definitely definitely write that test

**Pull request merged**